put_message_on_bucket.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+import json
+
+from google.cloud import pubsub_v1, storage
+from termcolor import colored
+
+
+def main(subscription_name, project_id, number_messages, id_search, action, bucket_name):
+    subscriber = pubsub_v1.SubscriberClient()
+    response = []
+
+    subscription_path = subscriber.subscription_path(
+        project_id, subscription_name)
+    for _ in range(number_messages):
+        message = subscriber.pull(subscription_path, max_messages=1, return_immediately=True)
+        if not message.received_messages:
+            break
+        response.extend(message.received_messages)
+    if not response:
+        print(colored('No messages on pubsub ', 'red'))
+        return
+    elif id_search:
+        matching_id_messages = [msg for msg in response if id_search == msg.message.message_id]
+        if matching_id_messages:
+            if action == 'MOVE':
+                client = storage.Client()
+                bucket = client.get_bucket(bucket_name)
+                pubsub_dict = {'message_id': matching_id_messages[0].message.message_id,
+                               'data': matching_id_messages[0].message.data.decode()}
+                bucket.blob(f'{subscription_name}-{matching_id_messages[0].message.message_id}')\
+                    .upload_from_string(json.dumps(pubsub_dict), content_type='application/json')
+                print(colored('uploaded message: ', 'red'), colored(f'{matching_id_messages[0].message.message_id}', 'white'))
+                subscriber.acknowledge(subscription_path, (msg.ack_id for msg in matching_id_messages))
```

Comment:
Should be okay to just ack the single message ID here?

Reply:
Yeah I consider it safe to acknowledge it here and get it off the topic unless anyone else has a problem with it. Once it's on the bucket, it can't be removed until you do it manually in the Storage UI

Reply:
Sorry, I mean just ack'ing the single message here (as we would only have one message) instead of looping through a list with one element...if that makes sense!